source/slang/slang-lower-to-ir.cpp

```diff
@@ -3546,6 +3556,21 @@ struct ExprLoweringContext
         // TODO: also need to handle this-type substitution here?
     }
 
+    void validateInvokeExprArgsWithFunctionModifiers(
+        InvokeExpr* expr,
+        FunctionDeclBase* decl,
+        List<IRInst*>& irArgs)
+    {
+        if (auto glslRequireShaderInputParameter = decl->findModifier<GLSLRequireShaderInputParameterAttribute>())
```

Comment:
Is this attribute something that the user will write? If not we can skip the validation.

Reply:
Invoke arguments may need to be validated that `in` objects are being used. Some functions requires `in` objects as a parameter